src/pybind/cephfs/cephfs.pyx

```diff
@@ -229,9 +228,12 @@ cdef make_ex(ret, msg):
     """
     ret = abs(ret)
     if ret in errno_to_exception:
-        return errno_to_exception[ret](ret, msg)
+        e = errno_to_exception[ret](ret, msg)
     else:
-        return OSError(ret, msg)
+        e = OSError(ret, msg)
+
+    e.errno = ceph_to_hostos_errno(e.errno)
```

Comment:
I think this should happen at line 230 above. Otherwise you're not using the hostos errno in the `errno_to_exception` table lookup.